frontend/public/components/resource-dropdown.tsx

```diff
@@ -92,51 +65,79 @@ const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = (props) => {
   const kinds = resources.groupBy((m) => m.kind);
   const isDup = (kind) => kinds.get(kind).size > 1;
 
-  const isKindSelected = (kind: string) => {
-    return _.includes(selected, kind);
-  };
-  // Create dropdown items for each resource.
-  const items = resources
-    .map((model) => (
-      <DropdownItem
-        key={referenceForModel(model)}
-        model={model}
-        showGroup={isDup(model.kind)}
-        checked={isKindSelected(referenceForModel(model))}
-      />
-    ))
-    .toJS();
-
-  const autocompleteFilter = (text, item) => {
-    const { model } = item.props;
-    if (!model) {
-      return false;
-    }
+  React.useEffect(() => {
+    setSelectedOptions(selected);
+  }, [selected]);
 
-    return fuzzy(
-      _.toLower(text),
-      model.labelKey ? _.toLower(t(model.labelKey)) : _.toLower(model.kind),
+  const items = [];
+  resources.forEach((model) => {
+    items.push(
+      <SelectOption
```

Comment:
Prefer `Array.map`.

```ts
const items = resources.map((model: K8sKind) => <SelectOption>...</SelectOption>;
```

Reply:
The result of the map should be the same... except TypeScript is able to infer a more specific type for the array. When you initialize the array as `const items = []`, TypeScript is treating it as an array of `any`, so we're skipping some type checking.

Can you share the code you tried?

Reply:
Here is the code I tried:
```
const items = resources
    .map((model: K8sKind) => (
      <SelectOption
        key={referenceForModel(model)}
        value={referenceForModel(model)}
        data-filter-text={`${model.abbr}${model.labelKey ? t(model.labelKey) : model.kind}`}
      >
        <span className="co-resource-item">
          <span className="co-resource-icon--fixed-width">
            <ResourceIcon kind={referenceForModel(model)} />
          </span>
          <span className="co-resource-item__resource-name">
            <span>
              {model.labelKey ? t(model.labelKey) : model.kind}
              {model.badge && model.badge === 'Tech Preview' && (
                <span className="co-resource-item__tech-dev-preview">
                  {t('public~Tech Preview')}
                </span>
              )}
            </span>
            {isDup(model.kind) && (
              <span className="co-resource-item__resource-api text-muted co-truncate show co-nowrap small">
                {model.apiGroup || 'core'}/{model.apiVersion}
              </span>
            )}
          </span>
        </span>
      </SelectOption>
    ))
```
Pretty straight forward, but It seems like PF really wants an actual array.

With that said, I think just changing it to an array using `.toArray()` might be  a better option than the approach what I was taking.  Updated commit coming soon.

Reply:
@kdoberst Ah, `resources` is not an array. It uses Immutable.js! I see now. Yes, you'll need `toArray`!